### ISSUE TITLE
bug: iosxr_router_bgp_vrf "neighbors" always bring a change when using dot notation with trailing zero

### ISSUE THREAD

**Issue by edudppaz** [CiscoDevNet/terraform-provider-iosxr#263]
When using a dot notation for the ASN for bgp neighbors, and the ASN has trailing zeros , for example: `65500.100` the value gets properly pushed to the devices, but gets stored in the state as `65500.1`, so on each plan, terraform brings a change to the value:

```
  # iosxr_router_bgp_vrf.fusion_bgp_vrf_inside["fn01"] will be updated in-place
  ~ resource "iosxr_router_bgp_vrf" "fusion_bgp_vrf_inside" {
        id                            = "Cisco-IOS-XR-um-router-bgp-cfg:/router/bgp/as[as-number=64599]/vrfs/vrf[vrf-name=abc]"
      ~ neighbors                     = [
          ~ {
              ~ remote_as                 = "64597.1" -> "64597.100"
                # (5 unchanged attributes hidden)
            },
        ]
        # (4 unchanged attributes hidden)
    }
```

It seems like the value is being saved as floating point or int on the state, it should be saved as a string to allow trailing zeros.

**Comment by kuba-mazurkiewicz** [comment]
Can you enable debugging and paste content of GET response ?

**Comment by danischm** [comment]
https://github.com/CiscoDevNet/terraform-provider-iosxr/commit/d1282fe70b6be203cfb8b05712993e9392813142

**Comment by danischm** [comment]
Workaround implemented in 0.5.2 release.

**Comment by danischm** [comment]
FYI, we have reverted the workaround now (https://github.com/CiscoDevNet/terraform-provider-iosxr/commit/b8ea0497df6ad33a8af28682a9ca25e1bea63bba), as recent versions of XR (e.g. 24.2.2) have fixed the encoding of dotted decimal AS numbers.
